xdsl/parser.py

```diff
@@ -271,11 +271,14 @@ def _find_token_end(self, start: int | None = None) -> int:
             if self.input.content.startswith(part, i):
                 return i + len(part)
         # Otherwise return the start of the next break
-        return min(
+        break_pos = list(
```

Comment:
```suggestion
        break_pos = [
            self.input.content.find(part, i) for part in self._break_on)
            if part
        ]
```

Reply:
Or actually you can leave it as the generator, and then `return min(min(break_pos), self.input.len))`

Reply:
Done!

Reply:
Actually no, it doesn't work, since the problem is that you cannot have an empty iterator in the `min`.
I switched back to the previous implementation, anyway the code is going to be deleted soon